python/pyarrow/parquet/core.py

```diff
@@ -1125,9 +1119,7 @@ def _get_pandas_index_columns(keyvalues):
 
 
 def _is_local_file_system(fs):
-    return isinstance(fs, LocalFileSystem) or isinstance(
-        fs, legacyfs.LocalFileSystem
-    )
+    return isinstance(fs, LocalFileSystem)
```

Comment:
Given we only use this in one place and it is shorter now, I would just inline the `isinstance(fs, LocalFileSystem)`